src/main/java/com/worldwarofants/game/util/RandGen.java

```diff
@@ -0,0 +1,77 @@
+package com.worldwarofants.game.util;
+
+import java.util.Random;
+
+/*
+    Utility class that provides an abstraction of Java's Random class.
+    Its goal is to provide developers with a simple way to work with
+    random number generation, as well as allow for precise tweaking
+    of exact methods used.
+
+    Please make sure to generate your random numbers within the project
+    exclusively using this package.
+*/
+
+public class RandGen extends java.util.Random {
+    private static final long serialVersionUID = -5378246343912387722L;
+
+    Random generator;
+    long seed;
+    String seedString; // If a custom text input has been provided by user and hashed,
+
+    // Overloaded seedless constructor
+    public RandGen() {
+        this.generator = new Random(); // Create a generator
+        this.seed = generator.nextLong(); // Ask it to generate a seed for itself
+        this.seedString = Long.toString(this.seed); // Save it to seedString as well
+        this.generator.setSeed(seed); // And set its seed to what it gave
+    }
+
+    // Overloaded seeded (with a string) constructor
+    public RandGen(String seedString) {
+        if (seedString.matches("-?\\d+(\\.\\d+)?")) { // If the provided string is a valid long, use it as the seed
```

Comment:
Put that regex in a constant with a good name, instead of writing the comment. Like this for example, at the top of the class: 
``` private static const String VALID_LONG_REGEX = "-?\\d+(\\.\\d+)?"; ```
Or however you think is better. And then when you use it, it'll look like this: 
``` if (seedString.matches(VALID_LONG_REGEX)) { ... } ```
It will read like actual English. See? No comments necessary here. The code says what it does on its own.

Reply:
Java uses final instead of const.
Nevertheless, improved.

Reply:
That's my work with Kotlin getting in the way, thanks for the reminder.